src/main/java/decodex/parser/Parser.java

```diff
@@ -1,5 +1,66 @@
 package decodex.parser;
 
+import decodex.commands.Command;
+import decodex.commands.ExitCommand;
+import decodex.data.DataManager;
+import decodex.modules.ModuleManager;
+import decodex.ui.Ui;
+
+import java.util.Arrays;
+
+/**
+ * Parses and validates the user input.
+ */
 public class Parser {
 
+    /**
+     * Returns the type of command that user has entered.
+     *
+     * @param userInput The input specified by the user.
+     * @return The type of command.
+     */
+    public String getCommandType(String userInput) {
+        String[] tokens = userInput.split(" ");
+        return tokens[0];

```

Comment:
What if `tokens` is an array with no elements? Perhaps do a check here and throw a `ParserException` if the length of `tokens` is not correct.